scipy/special/orthogonal.py

```diff
@@ -525,12 +532,277 @@ def h_roots(n, mu=False):
+    """Lemma 3.1 (Tricomi initial guesses)
+    """
+    tauk = compute_tauk(n, k)
+    sigk = cos(0.5*tauk)**2
+    a = n % 2 - 0.5
+    nu = 4.0*floor(n/2.0) + 2.0*a + 2.0
+    # Initial approximation of Hermite roots (square)
+    xksq = nu*sigk - 1.0/(3.0*nu) * (5.0/(4.0*(1.0-sigk)**2) - 1.0/(1.0-sigk) - 0.25)
+    return xksq
+
+
+def compute_am(m):
+    """Compute approximations to the roots of the Airy Ai(x) function.
+    This formula is exact for m > 10.
+    """
+    sm = 3*pi*(4*m-1) / 8.0
+    coeffs = row_stack([1.0, 5.0/48.0, -5.0/36.0, 77125.0/82944.0,
+                        -108056875.0/6967296.0, 162375596875.0/334430208.0])
+    smp = column_stack([ones_like(sm), sm**(-2), sm**(-4), sm**(-6), sm**(-8), sm**(-10)])
+    am = -sm**(2.0/3.0) * dot(smp, coeffs).reshape((-1,))
+    return am
+
+
+def airy_root(m):
+    """Compute approximations to all roots of the Airy Ai(x) function.
+    """
+    airyroots = array([
+            -2.3381074104597670385,
+             -4.0879494441309706166,
+             -5.5205598280955510591,
+             -6.7867080900717589988,
+             -7.9441335871208531231,
+             -9.0226508533409803802,
+             -10.040174341558085931,
+             -11.008524303733262893,
+             -11.936015563236262517,
+             -12.828776752865757200])
+
+    am = zeros_like(m, dtype=floating)
+    I = m <= 10
+    am[I] = airyroots[m[I]-1]
+    J = m > 10
+    am[J] = compute_am(m[J])
+    return am
+
+
+def initial_nodes_b(n, k):
+    """Lemma 3.2 (Gatteschi initial guesses)
+    """
+    a = n % 2 - 0.5
+    nu = 4.0*floor(n/2.0) + 2.0*a + 2.0
+    # Airy roots by approximation
+    ak = airy_root(k)
+    # Initial approximation of Hermite roots (square)
+    xksq = (nu +
+            2.0**(2.0/3.0) * ak * nu**(1.0/3.0) +
+            1.0/5.0 * 2.0**(4.0/3.0) * ak**2 * nu**(-1.0/3.0) +
+            (9.0/140.0 - 12.0/175.0 * ak**3) * nu**(-1.0) +
+            (16.0/1575.0 * ak + 92.0/7875.0 * ak**4) * 2.0**(2.0/3.0) * nu**(-5.0/3.0) -
+            (15152.0/3031875.0 * ak**5 + 1088.0/121275.0 * ak**2) * 2.0**(1.0/3.0) * nu**(-7.0/3.0))
+    return xksq
+
+
+def initial_values(n):
+    """Compute initial approximetaion to the Hermite roots of order n.
+
+    Both initial guesses are used for the region where they are
+    accurate and then combined.
+    """
+    # Turnover point
+    # linear polynomial fit to error of 10, 25, 40, ..., 1000 point rules
+    fit = 0.49082003*n - 4.37859653
+    turnover = around(fit).astype(int)
+    # Compute all approximations
+    ia = arange(1, int(floor(n*0.5)+1))
+    ib = flipud(arange(1, int(1+n-ceil(n*0.5))))
+    xasq = initial_nodes_a(n, ia[:turnover+1])
+    xbsq = initial_nodes_b(n, ib[turnover+1:])
+    # Combine
+    iv = sqrt(hstack([xasq, xbsq]))
+    # Central node is always zero
+    if n % 2 == 1:
+        iv = hstack([0.0, iv])
+    return iv
+
+
+def pbcf(n, theta):
+    """Asymptotic series expansion of parabolic cylinder function.
```

Comment:
Does our parabolic cylinder function have this expansion?  I haven't looked or tried, but do you think it would be worth adding?  Could be a separate PR.